### PR TITLE
feat PUD-1302 (pci.storage.databases): add kafka connect engine base

### PR DESCRIPTION
**Pull request by jperchoc**
PUD-1302

Signed-off-by: Jonathan Perchoc <jonathan.perchoc@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/aiven-ga2`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #PUD-1302
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Add Kafka connect engine base.

## Related
```
aiven-ga2 <-- PUD-1302 <-- PUD-1303 <-- PUD-1306
                             ^
                             |-- PUD-1304 <-- PUD-1305 <-- PUD-1307 
```
[aiven-ga2 - new engines](https://github.com/ovh/manager/pull/6410)
[PUD-1302 - Add kafka connect engine base](https://github.com/ovh/manager/pull/6576)
[PUD-1303 - Add kafka connect connectors tab](https://github.com/ovh/manager/pull/6577)
[PUD-1304 - Add kafka connect add connector page](https://github.com/ovh/manager/pull/6578)
[PUD-1305 - Add kafka connect edit connector page ](https://github.com/ovh/manager/pull/6579)
[PUD-1306 - Add kafka connect delete connector page](https://github.com/ovh/manager/pull/6581)
[PUD-1307 - Add kafka connect connector tasks page](https://github.com/ovh/manager/pull/6582)

### CODE REVIEW

File: packages/manager/modules/pci/src/components/project/storages/databases/connectorConfiguration.class.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  getTransformFields(type) {
+    return this.transformConfiguration[type];
+  }
+
+  isExtra(field) {
+    const isFieldInConfig = this.getField(field);
+    const isFieldTransform = field.startsWith(TRANSFORM_PROPERTY_KEY);
+    return !(isFieldInConfig || isFieldTransform);
```
Comment (radireddy) [inline packages/manager/modules/pci/src/components/project/storages/databases/connectorConfiguration.class.js:59]:
```suggestion
    return !isFieldInConfig && !isFieldTransform;
```
Nitpick: Better readability

File: packages/manager/modules/pci/src/components/project/storages/databases/connectors.constants.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export const CONNECTOR_STATUS = {
```
Comment (radireddy) [inline packages/manager/modules/pci/src/components/project/storages/databases/connectors.constants.js:1]: 👍

**Review by radireddy — APPROVED** [review]

File: packages/manager/modules/pci/src/components/project/storages/databases/connectorConfiguration.class.js (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/connectorConfiguration.class.js]: Changed here: https://github.com/ovh/manager/pull/6576/commits/c0fc6fbff741f2e2fc3da451e66ca01f05553873

File: packages/manager/modules/pci/src/components/project/storages/databases/connectors.constants.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  BOOLEAN: 'boolean',
+  CLASS: 'class',
+  STRING: 'string',
+  LIST: 'list',
+};
+
+export const CLASS_PROPERTY_AS_STRING = ['header.converter'];
+
+export const TRANSFORM_PROPERTY = 'transforms';
+export const EXTRA_CONFIG_PROPERTY = '*';
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/components/project/storages/databases/connectors.constants.js:37]: **todo**: Better to add an `export default`

Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/connectors.constants.js:37]: Done here: https://github.com/ovh/manager/pull/6576/commits/4f7622e8b1c5a54ff77e71a6974f1bceae060d45

**Review by anooparveti — APPROVED** [review]

**Pull request merged**